### PR TITLE
Adding the multiplexor service publisher

### PR DESCRIPTION
**Pull request by joselsegura**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #[CCXDEV-12639](https://issues.redhat.com/browse/CCXDEV-12639)

## Type of change

- New feature (non-breaking change which adds functionality)
- Unit tests (no changes in the code)

## Testing steps

Tested locally withing the previous PR #175 

## Checklist
* [x] `pre-commit run -a` passes
* [ ] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [x] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by epapbak — APPROVED** [review]

File: ccx_messaging/publishers/multiplexor_kafka_publisher.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                topic, gzip.compress(outgoing_message, compresslevel=BEST_COMPRESSION)
+            )
+        else:
+            self.producer.produce(topic, outgoing_message)
+        self.producer.poll(0)
+
+    def publish(self, input_msg: dict[str, Any], response: set[str]) -> None:
+        """Check to which topic should the `input_msg` be sent."""
+        # Clean up of the input_msg, removing the unwanted additions
+        cleanup_input(input_msg)
```
Comment (juandspy) [inline ccx_messaging/publishers/multiplexor_kafka_publisher.py:89]: why is this needed?

File: ccx_messaging/publishers/multiplexor_kafka_publisher.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+        for mark in response:
+            topic = self.topics_mapping.get(mark)
+            if not topic:
```
Comment (juandspy) [inline ccx_messaging/publishers/multiplexor_kafka_publisher.py:93]: if a mapping is not found, shouldn't we treat it as IO by default? if the archive is not valid, the rule service will notice it, but we still need it on Ceph, right?

File: ccx_messaging/publishers/multiplexor_kafka_publisher.py (hunk 1 above)
Comment (joselsegura) [reply ccx_messaging/publishers/multiplexor_kafka_publisher.py:89]: The current consumers are adding some values like the topic, offset and partition to `input_msg`. Those values are not in the original message.

I need to find a way to do it better and see why was it _needed, so when everything is working expect some changes in the consumers in order to keep the `input_msg` as it was received. In the meantime, the quickest way to fix this is cleaning up the dictionary, if it contains those keys.

File: ccx_messaging/publishers/multiplexor_kafka_publisher.py (hunk 2 above)
Comment (joselsegura) [reply ccx_messaging/publishers/multiplexor_kafka_publisher.py:93]: It will depend on the configuration. By default, the engine will mark with "DEFAULT" every archive that doesn't match any defined rule, so it will be easy to configure. I prefer to don't force the implementation because one use case (currently, the only one, I know).

So, for the multiplexor instance that we will need for internal data pipeline we will have a mapping like

```
{
    "OLS": "archive-ols-new",
    "DEFAULT": "archive-io-new"
}
```

**Review by Bee-lee — APPROVED** [review]

File: ccx_messaging/publishers/multiplexor_kafka_publisher.py (hunk 1 above)
Comment (juandspy) [reply ccx_messaging/publishers/multiplexor_kafka_publisher.py:89]: instead of deleting parts, shouldn't we keep just the ones we care about? i.e. the archive path

**Review by juandspy — APPROVED** [review]
